src/com/amazon/ionelement/impl/LongIntElementImpl.kt

```diff
@@ -46,11 +46,12 @@ internal class LongIntElementImpl(
     override fun writeContentTo(writer: IonWriter) = writer.writeInt(longValue)
     override fun equals(other: Any?): Boolean {
         if (this === other) return true
-        if (javaClass != other?.javaClass) return false
+        if (other !is IntElement) return false
 
-        other as LongIntElementImpl
-
-        if (longValue != other.longValue) return false
+        when (other.integerSize) {
```

Comment:
🗺️ If both `IntElement`s are `IntElementSize.LONG`, then it's slightly more efficient to compare using `longValue` instead of `bigIntegerValue`.